### PR TITLE
Update:  Batches in packages tests

### PR DESCRIPTION
**Pull request by rmeena840**

### CODE REVIEW

File: .github/workflows/01-main.yaml [hunk 1]
```diff
@@ -47,5 +47,77 @@ jobs:
...
+      - name: Setup Node
+        uses: actions/setup-node@v2-beta
+        with:
+          node-version: "16"
+          cache: "yarn"
+      - name: Yarn version check
+        run: yarn --version
+      - name: Install dependencies
+        run: yarn install
+      - name: Execute format
```
[the remark this answers is not in the document]
Comment (rmeena840) [reply .github/workflows/01-main.yaml:67]: It also checks for linter as well, which is needed.

**Review by sczembor — COMMENTED** [review]
its ok but I think we should build the entire project once and run all the checks and then just execute the tests in batches. Is it possible or every batch is running in isolated environment?

**Comment by rmeena840** [comment]
> its ok but I think we should build the entire project once and run all the checks and then just execute the tests in batches. Is it possible or every batch is running in isolated environment?

Every batch is running in isolated env.

File: .github/workflows/01-main.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (sczembor) [reply .github/workflows/01-main.yaml:67]: If its possible lets move it to a one script and just invoke it for every batch so we do not repeat the same code multiple times

**Comment by rmeena840** [comment]
@sczembor Ready for another review.

File: .github/workflows/reusable-workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Node
+        uses: actions/setup-node@v2-beta
+        with:
+          node-version: "16"
+          cache: "yarn"
+      - name: Yarn version check
```
Comment (sczembor) [inline .github/workflows/reusable-workflow.yaml:23]:
```suggestion
      - name: Check Yarn version
```

File: .github/workflows/reusable-workflow.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        with:
+          python-version: "3.10"
+          cache: "pipenv"
+      - name: Install poetry
+        run: pip install -U poetry
+      - name: Install pipenv
+        run: python -m pip install pipenv
+      - name: Install python dependencies
+        run: pipenv sync
+      - name: Run test
```
Comment (sczembor) [inline .github/workflows/reusable-workflow.yaml:44]:
```suggestion
      - name: Run tests
```

File: .github/workflows/reusable-workflow.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
+name: Reusable workflow example
```
Comment (sczembor) [inline .github/workflows/reusable-workflow.yaml:1]: why is it call example?

**Review by sczembor — COMMENTED** [review]
left few comments

File: .github/workflows/reusable-workflow.yaml (hunk 4 above)
Comment (rmeena840) [inline .github/workflows/reusable-workflow.yaml:1]:
```suggestion
name: Reusable workflow
```

Comment (rmeena840) [reply .github/workflows/reusable-workflow.yaml]: Corrected it.

**Review by sczembor — APPROVED** [review]
Good job! Please update the update the changelog before merging

**Comment by sczembor** [comment]
You did not update the change log

**Pull request merged**